src/zrc/api/validators.py

```diff
@@ -85,3 +90,23 @@ class HoofdzaakValidator:
     def __call__(self, obj: models.Model):
         if obj.hoofdzaak_id is not None:
             raise serializers.ValidationError(self.message, code=self.code)
+
+
+class RelevanteAndereZaakValidator:
+    def __call__(self, obj):
+        if sorted(obj.keys()) != ['aard_relatie', 'zaak']:
```

Comment:
this is too complicated - fields and their validators are intended for scalar values (barring some exceptions), complex things should be build as their own serializer as remarked above.